src/lib/homestore.cpp

```diff
@@ -208,17 +212,18 @@ void HomeStore::do_start() {
 
     if (has_index_service()) { m_index_service->start(); }
 
-    if (has_data_service()) {
-        m_data_service->start();
-    } else if (has_repl_data_service()) {
-        m_data_service->start();
-        s_cast< GenericReplService* >(m_repl_service.get())->start();
+    if (has_repl_data_service()) {
+        s_cast< GenericReplService* >(m_repl_service.get())->start(); // Replservice starts logstore & data service
+    } else {
+        if (has_data_service()) { m_data_service->start(); }
+        if (has_log_service() && inp_params.auto_recovery) {
+            // In case of custom recovery, let consumer starts the recovery and it is consumer module's responsibilities
+            // to start log store
+            m_log_service->start(is_first_time_boot() /* format */);
+        }
     }
 
-    // In case of custom recovery, let consumer starts the recovery and it is consumer module's responsibilities
-    // to start log store
-    if (has_log_service() && inp_params.auto_recovery) { m_log_service->start(is_first_time_boot() /* format */); }
-
+    m_cp_mgr->start_timer();
```

Comment:
I don't think CP itself is providing any service as such, it is an internal requirement which upper layer can tag along. Not sure it will buy us much.